### PR TITLE
Fix/update recovery machine UTs and enable them

### PR DESCRIPTION
**Pull request by t7ko-seagate**
# Problem Statement
- Recovery Machine UTs were commented out some time ago.  Need to uncomment, make sure they build and run OK, fix what is needed.  Also, need to modify these tests to test server-server scenario, not just server-client.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: dtm0/recovery.c [hunk 1]
```diff
@@ -1834,24 +1836,14 @@ static void local_recovery_fom_coro(struct m0_fom *fom)
...
-		/* Wait until the moment where we should start recovery. */
-		do {
-			M0_CO_FUN(CO(fom), heq_await(fom, &F(state), &F(eoq)));
-			if (F(eoq))
-				goto out;
-
-			if (F(state) == M0_NC_ONLINE && !ALL2ALL)
-				break;
-		} while (F(state) != M0_NC_DTM_RECOVERING);
+	F(recovered) = rf->rf_is_volatile;
```
Comment (madhavemuri) [inline dtm0/recovery.c:1839]: Why is_mkfs() is removed from here ?

Comment (t7ko-seagate) [reply dtm0/recovery.c:1838]: This fix was made after thorough discussion with IvanA.

Reasons:

1. Unit test thinks it runs in mkfs mode (`cs_ctx->cc_mkfs` is true).  So this logic was breaking recovery machine unit tests.
2. This is_mkfs() check was added "blindly", on early stages, and never completely validated.
3. ==> So, this code was added "for future" and is breaking unit tests.  We discussed and decision was -- remove it now, and add proper code "in future" when the need arises and we know exactly what is needed and for which purpose.
4. After removal of this fix, all2all test (which also runs mkfs) succeeded, with no issues, which proves that fix is not breaking anything.
5. Treating "recovery during mkfs" is definitely future, not PI7.  And it is not yet defined and designed and researched.

**Review by madhavemuri — APPROVED** [review]

**Comment by cortx-admin** [comment]
<h1>Jenkins CI Result : <a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254/>Motr#1254</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>17</td><td><details><summary>:file_folder:</summary><p><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/43m0crate target=_blank>01motr-single-node/43m0crate</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/52motr-singlenode-sanity target=_blank>01motr-single-node/52motr-singlenode-sanity</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/53clusterusage-alert target=_blank>02motr-single-node/53clusterusage-alert</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/41motr-conf-update target=_blank>02motr-single-node/41motr-conf-update</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/61sns-repair-motr-1n-1f target=_blank>03motr-single-node/61sns-repair-motr-1n-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/69sns-repair-motr-quiesce target=_blank>03motr-single-node/69sns-repair-motr-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/62sns-repair-motr-mf target=_blank>03motr-single-node/62sns-repair-motr-mf</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/70sns-failure-after-repair-quiesce target=_blank>03motr-single-node/70sns-failure-after-repair-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/63sns-repair-motr-1k-1f target=_blank>03motr-single-node/63sns-repair-motr-1k-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/60sns-repair-motr-1f target=_blank>03motr-single-node/60sns-repair-motr-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/66sns-repair-motr-abort-quiesce target=_blank>03motr-single-node/66sns-repair-motr-abort-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/24motr-dix-repair-lookup-insert-spiel target=_blank>03motr-single-node/24motr-dix-repair-lookup-insert-spiel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/68sns-repair-motr-shutdown target=_blank>03motr-single-node/68sns-repair-motr-shutdown</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/64sns-repair-motr-ios-fail target=_blank>03motr-single-node/64sns-repair-motr-ios-fail</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/24motr-dix-repair-lookup-insert-m0repair target=_blank>03motr-single-node/24motr-dix-repair-lookup-insert-m0repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/65sns-repair-motr-abort target=_blank>03motr-single-node/65sns-repair-motr-abort</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>15</td><td><details><summary>:file_folder:</summary><p><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/28sys-kvs target=_blank>01motr-single-node/28sys-kvs</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/04initscripts target=_blank>01motr-single-node/04initscripts</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/51kem target=_blank>02motr-single-node/51kem</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/20rpc-session-cancel target=_blank>02motr-single-node/20rpc-session-cancel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/10pver-assign target=_blank>02motr-single-node/10pver-assign</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/21fsync-single-node target=_blank>02motr-single-node/21fsync-single-node</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/13dgmode-io target=_blank>02motr-single-node/13dgmode-io</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/14poolmach target=_blank>02motr-single-node/14poolmach</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/11m0t1fs target=_blank>02motr-single-node/11m0t1fs</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/26motr-user-kernel-tests target=_blank>02motr-single-node/26motr-user-kernel-tests</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/08spiel target=_blank>02motr-single-node/08spiel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/06conf target=_blank>03motr-single-node/06conf</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/36spare-reservation target=_blank>03motr-single-node/36spare-reservation</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>10</td><td><details><summary>:file_folder:</summary><p><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/05confgen target=_blank>01motr-single-node/05confgen</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/06hagen target=_blank>01motr-single-node/06hagen</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/01net target=_blank>01motr-single-node/01net</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/03console target=_blank>01motr-single-node/03console</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/01motr-single-node/02rpcping target=_blank>01motr-single-node/02rpcping</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/07m0d-fatal target=_blank>02motr-single-node/07m0d-fatal</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/02motr-single-node/67fdmi-plugin-multi-filters target=_blank>02motr-single-node/67fdmi-plugin-multi-filters</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/08spiel-multi-confd target=_blank>03motr-single-node/08spiel-multi-confd</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit/03motr-single-node/04sss target=_blank>03motr-single-node/04sss</a></p></details></td></tr><tr><td>Total</td><td>42</td><td><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1254//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

**Comment by ivan-alekhin** [comment]
merged as https://github.com/Seagate/cortx-motr/commit/a94ebc7de2a88896485928674c53c4d6c604ada3 "dtm0/recovery: Enable and update recovery UTs."

**Comment by t7ko-seagate** [comment]
Merged -- see https://github.com/Seagate/cortx-motr/pull/1772#issuecomment-1142345268

Closing the PR.

**Pull request closed without merging**